### PR TITLE
Ra 138 display new fields on pipeline detail pages

### PR DESCRIPTION
**Pull request by JeremyGibson**
Adds the a details block for pipelines to the project details template. This also adds a migration from the 'project_capacity_unit' field from an integer to a slugified string, and simplifies the CSV export view to remove one of the instances of a CASE statement.

### CODE REVIEW

File: infrastructure/migrations/0044_auto_20190301_0746.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    atomic = False
+
+    dependencies = [
+        ('infrastructure', '0043_infrastructuretype_show_on_map'),
+    ]
+
+    # The VIEW must be dropped before the AlterField, to account for database constraints on
+    # altering a table upon which the VIEW depends. It will be rebuilt in the last operation.
+    operations = [
+        migrations.RunSQL("DROP VIEW infrastructure_projects_export_view"),
```
Comment (copelco) [inline infrastructure/migrations/0044_auto_20190301_0746.py]: 🆒

File: infrastructure/export.py [hunk 2]
```diff
@@ -262,10 +261,6 @@ def refresh_views():
                     END
                     "project_CO2_emissions_unit",
                     project_capacity,
-                    CASE
```
Comment (copelco) [inline infrastructure/export.py]: 💯

File: infrastructure/models.py [hunk 3]
```diff
@@ -400,6 +433,31 @@ def fuzzy_planned_completion_date(self):
             self.planned_completion_month,
             self.planned_completion_day
         )
+
+    @property
+    def pipeline_capacity_property(self):
```
Comment (copelco) [inline infrastructure/models.py]: These ``_property`` functions feel like template display logic to me. Is there a reason why this is in the model and not in the template?

File: infrastructure/models.py [hunk 4]
```diff
@@ -323,11 +357,10 @@ class Project(Publishable):
...
-    project_capacity = models.FloatField(blank=True, null=True, help_text="MW")
+    project_capacity = models.FloatField(blank=True, null=True)
 
-    project_capacity_unit = models.PositiveSmallIntegerField(
-        blank=True, null=True, choices=ProjectPlantUnits.UNITS
-    )
+    project_capacity_unit = models.CharField(blank=True, default='', max_length=20,
+                                             choices=ProjectCapacityUnits.UNITS)
 
     project_capacity_timeframe = models.CharField(blank=True, max_length=10,
```
Comment (copelco) [inline infrastructure/models.py]: Hmm, looking at the mockup, should this include **annually** as an option? Right now we can only display **3.8 tons per year**

File: infrastructure/models.py [hunk 5]
```diff
@@ -400,6 +433,31 @@ def fuzzy_planned_completion_date(self):
...
             self.planned_completion_day
         )
+
+    @property
+    def pipeline_capacity_property(self):
+        if self.project_capacity is None:
+            return None
+        pc = str(self.project_capacity)
+        if self.project_capacity_unit:
+            pc += " {}".format(ProjectCapacityUnits.get_human(self.project_capacity_unit))
```
Comment (copelco) [inline infrastructure/models.py]: I think we should lower the casing. Currently we show **3.8 Tons per year** as opposed to **3.8 tons per year** (which is in the PDF)

File: infrastructure/templates/infrastructure/project_detail.html [hunk 6]
```diff
@@ -217,6 +217,50 @@ <h3>Substation(s)</h3>
...
+                        {% endif %}
+                     </p>
+                  </section>
+               </div>
+               <div class="row double">
+                  <section>
+                     <h3>Fuel(s)</h3>
+                     <ul class="clean">
+                        {% for fuel in fuel_types %}
+                           <li>{{ fuel }}</li>
```
Comment (copelco) [inline infrastructure/templates/infrastructure/project_detail.html]: I wonder if these are supposed to be categories.. hard to tell from the wireframe, but a generic **Oil** item seems broad.

Comment (copelco) [reply infrastructure/templates/infrastructure/project_detail.html]: Also, I think we should add an empty case with a dash:

```
{% for ... %}
blah
{% empty %}
  <li>-</li>
{% endfor %}
```

File: infrastructure/models.py (hunk 3 above)
Comment (JeremyGibson) [reply infrastructure/models.py]: To me it seemed just a little more elegant, I'm more than happy to move this into the template though.

File: infrastructure/models.py (hunk 4 above)
Comment (JeremyGibson) [reply infrastructure/models.py]: The mock-up shows annually but the pick lists we were given do not have that as an option.  I just went with the pick lists.

File: infrastructure/models.py (hunk 5 above)
Comment (JeremyGibson) [reply infrastructure/models.py]: This is another case of "this is the format in the pick list that they gave us" so I thought best to follow it. I agree, lower case looks better. If we do lower case, should we lower case only in the template view, but  keep the upper case in the admin?

File: infrastructure/templates/infrastructure/project_detail.html (hunk 6 above)
Comment (JeremyGibson) [reply infrastructure/templates/infrastructure/project_detail.html]: done

File: infrastructure/models.py (hunk 3 above)
Comment (JeremyGibson) [reply infrastructure/models.py]: Based on slack convo. I'm going to leave this as is.

File: infrastructure/templates/infrastructure/project_detail.html (hunk 6 above)
Comment (copelco) [reply infrastructure/templates/infrastructure/project_detail.html]: Is it? I don't see this change.

File: power_plant_import/tests/test_import_to_database.py [hunk 7]
```diff
@@ -420,10 +421,11 @@ def test_powerplants_projects_created(self):
...
         )
         self.assertEqual(project_ouessant1.project_capacity, 0.5)
-        self.assertEqual(
-            project_ouessant1.project_capacity_unit,
-            {value.upper(): key for key, value in ProjectPlantUnits.UNITS}['MW']
-        )
+        if project_ouessant1.project_capacity_unit != '':
+            self.assertEqual(
+                project_ouessant1.project_capacity_unit,
+                ProjectCapacityUnits.get_human(project_ouessant1.project_capacity_unit)
```
Comment (copelco) [inline power_plant_import/tests/test_import_to_database.py]: Does this line still work?

Comment (JeremyGibson) [reply power_plant_import/tests/test_import_to_database.py]: :man_facepalming:

**Pull request merged**